gtsam/linear/AcceleratedPowerMethod.h

```diff
@@ -59,66 +60,68 @@ class AcceleratedPowerMethod : public PowerMethod<Operator> {
   // Constructor from aim matrix A (given as Matrix or Sparse), optional intial
   // vector as ritzVector
   explicit AcceleratedPowerMethod(
-      const Operator &A, const boost::optional<Vector> initial = boost::none)
+      const Operator &A, const boost::optional<Vector> initial = boost::none,
+      const double initialBeta = 0.0)
       : PowerMethod<Operator>(A, initial) {
-    Vector x0;
-    // initialize ritz vector
-    x0 = initial ? Vector::Random(this->dim_) : initial.get();
-    Vector x00 = Vector::Random(this->dim_);
-    x0.normalize();
-    x00.normalize();
-
     // initialize Ritz eigen vector and previous vector
-    previousVector_ = update(x0, x00, beta_);
-    this->ritzVector_ = update(previousVector_, x0, beta_);
-    this->perturb();
-
-    // set beta
-    Vector init_resid = this->ritzVector_;
-    const double up = init_resid.transpose() * this->A_ * init_resid;
-    const double down = init_resid.transpose().dot(init_resid);
-    const double mu = up / down;
-    beta_ = mu * mu / 4;
-    setBeta();
+    this->ritzVector_ = initial ? initial.get() : Vector::Random(this->dim_);
+    this->ritzVector_.normalize();
+    previousVector_ = Vector::Zero(this->dim_);
+
+    // initialize beta_
+    if (!initialBeta) {
+      estimateBeta();
+    } else {
+      beta_ = initialBeta;
+    }
   }
 
-  // Update the ritzVector with beta and previous two ritzVector
-  Vector update(const Vector &x1, const Vector &x0, const double beta) const {
+  // Update the ritzVector with beta and previous two ritzVector, and return
```

Comment:
This docstring **still** does not accurately describe what this function does!  [This function doesn't "**update**" ] anything!